zebra-state/src/sled_state.rs

```diff
@@ -104,23 +170,41 @@ impl FinalizedState {
             &self.hash_by_height,
             &self.height_by_hash,
             &self.block_by_height,
+            &self.utxo_by_outpoint,
         )
-            .transaction(move |(hash_by_height, height_by_hash, block_by_height)| {
-                // TODO: do serialization above
-                // for some reason this wouldn't move into the closure (??)
-                let block_bytes = block
-                    .zcash_serialize_to_vec()
-                    .expect("zcash_serialize_to_vec has wrong return type");
-
-                // TODO: check highest entry of hash_by_height as in RFC
-
-                hash_by_height.insert(&height_bytes, &hash.0)?;
-                height_by_hash.insert(&hash.0, &height_bytes)?;
-                block_by_height.insert(&height_bytes, block_bytes)?;
-
-                // for some reason type inference fails here
-                Ok::<_, sled::transaction::ConflictableTransactionError>(hash)
-            })
+            .transaction(
+                move |(hash_by_height, height_by_hash, block_by_height, utxo_by_outpoint)| {
+                    // TODO: do serialization above
+                    // for some reason this wouldn't move into the closure (??)
+                    let block_bytes = block
+                        .zcash_serialize_to_vec()
+                        .expect("zcash_serialize_to_vec has wrong return type");
+
+                    // TODO: check highest entry of hash_by_height as in RFC
```

Comment:
Is there a ticket or checklist item for this TODO?

Reply:
no idea, the comment predates this PR

Reply:
I wrote that comment when implementing the most basic version of the sled state that would unblock the sync work.  It should be on the list of things to implement to complete the sled state, along with the other TODOs in the source code below.